isic_challenge_scoring/classification.py

```diff
@@ -61,9 +69,36 @@ def __init__(
         )
 
         self.overall = self.aggregate.at['balanced_accuracy']
-        self.validation = metrics.balanced_multiclass_accuracy(
-            truth_probabilities, prediction_probabilities, truth_weights.validation_weight
-        )
+
+        if validation_metric:
+            if validation_metric == ValidationMetric.BALANCED_ACCURACY:
+                self.validation = metrics.balanced_multiclass_accuracy(
+                    truth_probabilities, prediction_probabilities, truth_weights.validation_weight
+                )
+            elif validation_metric == ValidationMetric.AVERAGE_PRECISION:
+                per_category_ap = pd.DataFrame(
```

Comment:
Why is this a `DataFrame`? The result of each call to `metrics.average_precision` results in a scalar, so a list of them should just be a `Series`.